rebasehelper/build_log_hooks/files_build_log_hook.py

```diff
@@ -0,0 +1,257 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import collections
+import difflib
+import fnmatch
+import os
+import re
+
+import six
+
+from rebasehelper.build_log_hook import BaseBuildLogHook
+from rebasehelper.helpers.macro_helper import MacroHelper
+from rebasehelper.logger import logger
+
+
+class FilesBuildLogHook(BaseBuildLogHook):
+    NAME = 'files'
+    CATEGORIES = None
+
+    # taken from build/files.c in RPM source
+    FILES_DIRECTIVES = {
+        '%artifact': None,
+        '%attr': None,
+        '%caps': None,
+        '%config': None,
+        '%defattr': None,
+        '%dev': None,
+        '%dir': None,
+        '%doc': '%{_defaultdocdir}/%{name}',
+        '%docdir': None,
+        '%dverify': None,
+        '%exclude': None,
+        '%ghost': None,
+        '%lang': None,
+        '%license': '%{_defaultlicensedir}/%{name}',
+        '%missingok': None,
+        '%pubkey': None,
+        '%readme': None,
+        '%verify': None,
+    }
+
+    @classmethod
+    def get_name(cls):
+        return cls.NAME
+
+    @classmethod
+    def get_categories(cls):
+        return cls.CATEGORIES
+
+    @classmethod
+    def _parse_build_log(cls, log_path, nvr):
+        """Parses a build log.
+
+        Args:
+            log_path (str): Path to the RPM build log.
+
+        Returns:
+            tuple: The first element is the type of error that was found
+                in the log (missing or deleted). The second element is
+                a list of problematic files.
+
+        """
+        try:
+            with open(log_path, 'r') as build_log:
+                lines = build_log.read().splitlines()
+        except IOError:
+            logger.error('There was an error opening %s', log_path)
+            return None, None
+
+        error_type = None
+        # Use set to avoid duplicate files.
+        files = set()
+
+        # Two types of error can occur. The file is either in the SPEC file but missing in sources
+        # or the other way around. Example lines of the first case:
+        # File not found: /root/rpmbuild/BUILDROOT/pello-0.1.1-1.fc27.x86_64/should/not/be/here
+        # File not found: /root/rpmbuild/BUILDROOT/pello-0.1.1-1.fc27.x86_64/should/not/be/here2
+        # Example lines of the second case:
+        # Installed (but unpackaged) file(s) found:
+        #   /usr/bin/pello
+        error_re = re.compile(
+            r'''
+            ^
+            (BUILDSTDERR:)?
+            \s*
+            (
+            (?P<missing>File \s not \s found:\s*)|
+            (?P<unpackaged>Installed \s \(but \s unpackaged\) \s file\(s\) \s found:)
```

Comment:
While this is not wrong, it's a little confusing (to me at least). I think you should left out those spaces around `\s`. And since you are matching general whitespace, you could as well use `\s+`.